### ISSUE TITLE
Can the fusion support for `functional.batch_norm` be added?

### ISSUE THREAD

**Issue by hoshibara** [pytorch/pytorch#122070]
### 🚀 The feature, motivation and pitch

I'm using [optimization.fuse](https://github.com/pytorch/pytorch/blob/e39aedfcc582eb897e5680c452215e66d5c59c78/torch/fx/experimental/optimization.py#L50) to fuse model's conv and batch_norm operations, but I found that this API can only match `nn.BatchNorm*d` but cannot match `functional.batch_norm`.
Can the fusion support for `functional.batch_norm` be added?

### Additional context

UT:
```python
import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.fx.experimental.optimization as optimization
from torch.profiler import profile, ProfilerActivity


class SimpleConvBNModel(nn.Module):
    def __init__(self, input_channels, output_channels, kernel_size, stride=1, padding=0):
        super(SimpleConvBNModel, self).__init__()
        self.conv = nn.Conv2d(input_channels, output_channels, kernel_size, stride, padding)
        self.num_features = output_channels
        self.gamma = nn.Parameter(torch.ones(self.num_features))
        self.beta = nn.Parameter(torch.zeros(self.num_features))
        self.bn2d = nn.BatchNorm2d(
            output_channels, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True
        )
        # Assuming running_mean and running_var are managed elsewhere or not used in this simplified example

    def forward(self, x):
        running_mean = torch.zeros(16).cuda()
        running_var = torch.ones(16).cuda()
        x = self.conv(x)

        # Case 1
        # Cannot be fused

        x = F.batch_norm(
            x,
            running_mean=running_mean,
            running_var=running_var,
            weight=self.gamma,
            bias=self.beta,
            training=False,
        )

        # Case 2
        # Can be fused
        # x = self.bn2d(x)
        return x


# Ensure everything is on the same device
model = SimpleConvBNModel(3, 16, 3).cuda()
model.eval()
model = optimization.fuse(model)


# Dummy input
input_tensor = torch.randn(4, 3, 64, 64).cuda()

# Forward pass
with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA]) as prof:
    with torch.no_grad():
        for i in range(100):
            output = model(input_tensor)
print(prof.key_averages().table(sort_by="self_cuda_time_total", max_name_column_width=100))
```
